### PR TITLE
feat: Add new field for the page meta data in templates

### LINKED ISSUE #14701: Add API to get page meta data
For templates, we need to display the list of pages and their ids. The templates API should store and return these meta data.

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
In order to import a specific page or browse to that page, the templates FE need page names and page ids. This PR replaces the list of page names with a list of objects that contains the page names and page ids.

Fixes #14701

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ApplicationTemplate.java [hunk 1]
```diff
@@ -21,7 +21,7 @@ public class ApplicationTemplate extends BaseDomain {
     private List<String> functions;
     private List<String> useCases;
     private List<String> datasources;
-    private List<String> pageNames;
+    private List<PageNameIdDTO> pages;
```
[the remark this answers is not in the document]
Comment (nayan-rafiq) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/dtos/ApplicationTemplate.java:24]: The templates are stored in cloud services. This DTO represents the template which is fetched from cloud services. Also the `pageNames` field was optional and was not used yet by FE so there is no chance of breaking anything. We're going to store some more information along with page name so `pageNames` was replaced by `pages`

**Review by AnaghHegde — APPROVED** [review]

**Pull request merged**